overlord/snapstate/snapstate.go

```diff
@@ -1890,23 +1890,30 @@ type RemoveFlags struct {
 // Remove returns a set of tasks for removing snap.
 // Note that the state must be locked by the caller.
 func Remove(st *state.State, name string, revision snap.Revision, flags *RemoveFlags) (*state.TaskSet, error) {
+	ts, _, err := removeTasks(st, name, revision, flags)
+	return ts, err
```

Comment:
This makes me wonder if we should also capture "snapshotSize" here and do the check for disk-space in a common helper between Remove and RemoveMany. Right now we have many disk-space checks when removing multiple snaps. Each snap will do a disk-space check and then we do one for the combined size. While not really a problem it's slightly odd and redundant. So maybe just calculating the estimated size in removeTasks and then do the diskspace check in a shared helper. This should also make the rewrite of the error unnecessary  because the requires total size and the snap names could be passed in. Hope this rambling makes sense, if not I'm happy to do some sketch work and push here or into a pastebin. Anyway, mostly a wondering if that would make things clearner (I stumbled on the unit tests a bit and I think that's the reason :)

Reply:
Thanks, that's a valid observation, I was wondering about this as well when working on this. I'll iterate a bit more on this aspect.

Reply:
Done.